From cee545f7e9d3e6d64fe42e6b469784ffbbd5eb02 Mon Sep 17 00:00:00 2001
From: "R, Kannappan" <r.kannappan@intel.com>
Date: Mon, 5 Oct 2015 19:32:20 +0530
Subject: [PATCH] sink_pe: respond to get_sink_cap at any state

sink port can respond to get_sink_cap request at any state.
source could request for this message after sending ps_rdy
command. if process to setup charging is delayed, this message
should not be ignored.

Change-Id: If5ac09115853dd3fcaa8d7ee9e687b9090f6c16a
Signed-off-by: R, Kannappan <r.kannappan@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-19347
---
 drivers/usb/typec/pd/sink_port_pe.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/usb/typec/pd/sink_port_pe.c b/drivers/usb/typec/pd/sink_port_pe.c
index 42bf4a4..99e827d 100644
--- a/drivers/usb/typec/pd/sink_port_pe.c
+++ b/drivers/usb/typec/pd/sink_port_pe.c
@@ -1025,9 +1025,8 @@ static int sink_port_policy_rcv_pkt(struct policy *p, struct pd_packet *pkt,
 			sink_handle_src_cap(sink, pkt);
 		break;
 	case PE_EVT_RCVD_GET_SINK_CAP:
-		if (sink->cur_state == PE_SNK_READY)
-			return snkpe_handle_give_snk_cap_state(sink);
-		break;
+		return snkpe_handle_give_snk_cap_state(sink);
+
 	case PE_EVT_RCVD_ACCEPT:
 	case PE_EVT_RCVD_REJECT:
 	case PE_EVT_RCVD_WAIT:
-- 
1.9.1

